File: merge-alternately/MergeAlternately.swift
class MergeAlternately {
    func mergeAlternately(_ word1: String, _ word2: String) -> String {
        var result = ""
        var i = 0, j = 0

        while i < word1.count, j < word2.count {
            if i == j {
                result.append(word1[word1.index(word1.startIndex, offsetBy: i)])
                i += 1
            } else {
                result.append(word2[word2.index(word2.startIndex, offsetBy: j)])
                j += 1
            }
        }

        while i < word1.count {
            result.append(word1[word1.index(word1.startIndex, offsetBy: i)])
            i += 1
        }

        while j < word2.count {
            result.append(word2[word2.index(word2.startIndex, offsetBy: j)])
            j += 1
        }

        return result
    }
}

File: merge-alternately/MergeAlternately.java
class MergeAlternately {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int i = 0, j = 0;

        while(i < word1.length() && j < word2.length()) {
            if(i == j) {
                sb.append(word1.charAt(i));
                i++;
            } else {
                sb.append(word2.charAt(j));
                j++;
            }
        }

        while(i < word1.length()) {
            sb.append(word1.charAt(i));
            i++;
        }

        while(j < word2.length()) {
            sb.append(word2.charAt(j));
            j++;
        }

        return sb.toString();
    }
}